### ISSUE TITLE
Wrong signature generated for SECP256K1 keys

### ISSUE THREAD

**Issue by davidatwhiletrue** [casper-ecosystem/casper-client-sdk#315]
https://github.com/casper-ecosystem/casper-js-sdk/blob/2219cbbc943b44f5562e1b74d4d4ecf1a89d201d/src/lib/Keys.ts#L636

Calling the method above returns a DER-encoded signature (70 bytes). Such signature is not valid for a deploy. To generate a r+s signature (64) bytes the method must be called with following options:

```
    const signature = secp256k1.signSync(
      sha256(Buffer.from(msg)),
      this.privateKey, {
          der: false
        }
    );
```

cc @hoffmannjan

**Comment by piotrwitek** [comment]
I think I'm impacted and blocked by this bug implementing Erc20 transfers when trying to sign the Erc20 transfer deploy, here is a deploy JSON and the error I got from the rpc so you can debug this issue.

TESTNET
Contract: https://testnet.cspr.live/contract-package/d3d63700c688e9fe5402b07cb413ddba6b8161b921ad31705ff098db22b8a99e

Deploy:
```
{"jsonrpc":"2.0","id":"0","method":"account_put_deploy","params":{"deploy":{"hash":"dec1baa53f2fa44ab13c381e9b72898ccc33d86ffbcdf2a4d64f2e73e3f40e12","header":{"account":"020304Bb523043dec125E3f31efA82fF2B8E30dF97211bcf87CDb3B6BaE6586AdD4A","timestamp":"2023-06-21T16:08:51.150Z","ttl":"30m","gas_price":1,"body_hash":"8c66c565a65a46f6f8620047cf592fb145fef873df6afac2baabb78ca9b345fe","dependencies":[],"chain_name":"casper-test"},"payment":{"ModuleBytes":{"module_bytes":"","args":[["amount",{"bytes":"0480f0fa02","cl_type":"U512"}]]}},"session":{"StoredContractByHash":{"hash":"d3d63700c688e9fe5402b07cb413ddba6b8161b921ad31705ff098db22b8a99e","entry_point":"transfer","args":[["recipient",{"bytes":"00f98f2ed4ea79f13cf3e5681e5b2dfbb2b9ef48770a76f04f2f934503522a44f5","cl_type":"Key"}],["amount",{"bytes":"0103","cl_type":"U256"}]]}},"approvals":[{"signer":"020304Bb523043dec125E3f31efA82fF2B8E30dF97211bcf87CDb3B6BaE6586AdD4A","signature":"02304402205fb8efb7766347460df408772aa6fef35d463f287d81c5022f96a006c30407b20220468ef62f5fd6d73d2bcbe47cef895d89324dc844a83c44b79ef55348cee532fc"}]}}}
```

Rpc error:
```
{
    "jsonrpc": "2.0",
    "id": "0",
    "error": {
        "code": -32602,
        "message": "Invalid params",
        "data": null
    }
}
```

As it is a blocker could you please update us on the priority of this bug please so we can plan our work around it? Thanks.

**Comment by hoffmannjan** [comment]
Hi guys, we should have fix ASAP and release it as bugfix, @gyroflaw is working on it. Will keep you in a loop

**Comment by piotrwitek** [comment]
Sounds amazing, thanks for a rapid response :)

**Comment by hoffmannjan** [comment]
Sorry, will close after release today :)

**Comment by hoffmannjan** [comment]
Released `2.13.2` - please update and confirm it works for you guys!

**Comment by piotrwitek** [comment]
I'll check tomorrow, thanks!

**Comment by RitaMAllenCA** [comment]
closed as released in 2.13.2
